src/libprojectM/MilkdropPresetFactory/MilkdropPreset.cpp

```diff
@@ -387,94 +394,55 @@ void MilkdropPreset::evaluateFrame()
 
 }
 
-void MilkdropPreset::initialize_PerPixelMeshes()
-{
-
-  int x,y;
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.cx_mesh[x][y]=presetOutputs().cx;
-	}}
-
-
-
-
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.cy_mesh[x][y]=presetOutputs().cy;
-	}}
-
-
-
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.sx_mesh[x][y]=presetOutputs().sx;
-	}}
-
-
-
-
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.sy_mesh[x][y]=presetOutputs().sy;
-	}}
-
-
-
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.dx_mesh[x][y]=presetOutputs().dx;
-	}}
-
-//std::cout<<presetOutputs().cx<<","<<presetOutputs().cy<<" "<<presetOutputs().dx<<","<<presetOutputs().dy<<std::endl;
-
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.dy_mesh[x][y]=presetOutputs().dy;
-	}}
-
-
 
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.zoom_mesh[x][y]=presetOutputs().zoom;
-	}}
-
-
-
-
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.zoomexp_mesh[x][y]=presetOutputs().zoomexp;
-	}}
-
-
-
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.rot_mesh[x][y]=presetOutputs().rot;
-	}}
-
-
-      for (x=0;x<presetInputs().gx;x++){
-	for(y=0;y<presetInputs().gy;y++){
-	  _presetOutputs.warp_mesh[x][y]=presetOutputs().warp;
-	}}
+#ifdef __SSE2__
+inline void init_mesh(float **mesh, const float value, const int gx, const int gy)
+{
+  __m128 mvalue = _mm_set_ps1(value);
```

Comment:
1. shouldn't the compiler do this stuff anyway?
2. what about memset?

Reply:
On one machine it seemed to make a big difference.  Though you'd think the compiler would do this. I haven't dug that deep into in gcc optimizations.  BTW, I've found gperftools really helpful.  Though I try to be careful to use builds with -DNDEBUG to make sure I'm profiling something meaningful.  Is there an easy way to do that using ./configure?

Reply:
I was profiling with xcode and Instruments. clang will emit different optimizations. Are you compiling with -O2/-O3/-Ofast?
and since this is a contiguous block of memory can we not just memset the region?

Reply:
I thought about using memset, but I don't think it will escalate to using 128bit writes (the compilers/libraries are very clever though).  I've just been compiling with the defaults, for profiling I just add -DNDEBUG to remove asserts().